src/pages/Productlisting.jsx

```diff
@@ -1,36 +1,65 @@
-import React from 'react'
-import Productcard from '../components/card/Productcard'
-import Footer from '../components/footer/Footer'
-import Filter from '../components/filter/Filter'
-import Navbar from '../components/navbar/Navbar'
-import { useProduct } from '../context/product-context'
-// import { useParams } from 'react-router-dom'
-
+import React from "react";
+import Productcard from "../components/card/Productcard";
+import Footer from "../components/footer/Footer";
+import Filter from "../components/filter/Filter";
+import Navbar from "../components/navbar/Navbar";
+import { useProduct } from "../context/product-context";
+import { useFilter } from "../context/filter-context";
+import { getSortedProduct } from "../utils/sort-by-price";
+import { getCategoryProduct } from "../utils/filter-by-category";
+import { getRatingProduct } from "../utils/filter-by-rating";
+import { getStockProduct } from "../utils/filter-by-stock";
+import { getSearchProduct } from "../utils/filter-by-search";
+import { getPricedProduct } from "../utils/filter-by-price";
```

Comment:
here as also you can use barrel exports and imports

Reply:
done